common/internal_model/src/main/scala/weco/catalogue/internal_model/identifiers/IdentifierType.scala

```diff
@@ -1,47 +1,106 @@
 package weco.catalogue.internal_model.identifiers
 
-import java.io.InputStream
+import enumeratum.{Enum, EnumEntry}
+import io.circe.{Decoder, Encoder}
+import weco.catalogue.internal_model.locations.LocationType.findValues
 
+import java.io.InputStream
 import scala.io.Source
 
-case class IdentifierType(
-  id: String,
-  label: String
-)
-
-case object IdentifierType {
-  private val stream: InputStream =
-    getClass.getResourceAsStream("/identifier-schemes.csv")
-  private val source = Source.fromInputStream(stream)
-  private val csvRows = source.mkString
-    .split("\n")
-
-  // identifier-schemes.csv is a list of 2-tuples, e.g.:
-  //
-  //
-  //    miro-image-number,Miro image number
-  //    wellcome-library-videodisk-number,Wellcome library videodisk number
-  //    sierra-system-number,Sierra system number
-  //
-  private val identifierTypeMap: Map[String, IdentifierType] = csvRows
-    .map { row =>
-      val columns = row.split(",").map(_.trim)
-      assert(columns.length == 2)
-      Map(
-        columns(0) -> IdentifierType(
-          id = columns(0),
-          label = columns(1)
-        ))
-    }
-    .fold(Map()) { (x, y) =>
-      x ++ y
-    }
-
-  def apply(platformId: String): IdentifierType =
-    identifierTypeMap.get(platformId) match {
-      case Some(id) => id
-      case None =>
-        throw new IllegalArgumentException(
-          s"Unrecognised identifier type: [$platformId]")
-    }
+sealed trait IdentifierType extends EnumEntry {
+  val id: String
+  val label: String
+}
+
+object IdentifierType extends Enum[IdentifierType] {
+  val values = findValues
+
+  implicit val locationTypeEncoder: Encoder[IdentifierType] =
+    Encoder.forProduct1("id")(_.id)
+
+  implicit val locationTypeDecoder: Decoder[IdentifierType] =
+    Decoder.forProduct1("id")(IdentifierType.withName)
+
+  def apply(id: String): IdentifierType =
+    IdentifierType.withName(id)
```

Comment:
How is it working without `entryName` like the other types use? Does it secretly support an `id` field as well as `entryName`?(!) https://github.com/wellcomecollection/catalogue/blob/51ab8cf0288a9362ceec65a318cfe6766534142a/common/internal_model/src/main/scala/weco/catalogue/internal_model/work/Format.scala#L10

Reply:
The answer might be… it doesn't 🙈 

I wanted to get this up for review quickly, so I haven't run the tests yet

Reply:
😬 well, there's your solution if it doesn't

Reply:
Indeed, that seems to be important…